scio-test/src/main/scala/com/spotify/scio/testing/SCollectionMatchers.scala

```diff
@@ -362,3 +362,10 @@ trait SCollectionMatchers {
   }
 
 }
+
+/** Helper class to ensure test data survives lambda serialization. */
+private class Externalizer[T: Coder](sc: ScioContext, original: T) extends Serializable {
```

Comment:
This should probably be public? Could be useful for user code lambda see/de cases.

Reply:
just wondering, since this implements Serializable, would it work to just implement the `writeObject`/`readObject` methods of Serializable?

Reply:
Talked IRL. Will get this in for now and do a separate PR for a public API.